be/src/common/config.h

```diff
@@ -288,6 +288,11 @@ CONF_mInt64(max_cumulative_compaction_num_singleton_deltas, "1000");
 CONF_Int32(cumulative_compaction_num_threads_per_disk, "1");
 // CONF_Int32(cumulative_compaction_write_mbytes_per_sec, "100");
 
+// This config is to limit the max candidate of compaction queue to avoid
+// too many candidates lead to OOM or cpu overload.
+// when candidate num reach this value, the condidate with lowest score will be dropped.
+CONF_mInt64(max_compaction_candidate_num, "40960");
```

Comment:
If you limit the compaction thread number, why it will consume much CPU even many candidates?

Reply:
this is a priority queue, push method complex is o(logn) and replace method complex is o(n).